Add Notary Project docs assessment (#27)
cncf/techdocs: assessments/0002-notary-project.md

# Notary Project Docs Assessment

Prepared by: `Nate Waddington`
Date: `2021-07-07`
Project: [Notary Project](https://github.com/notaryproject/)

## Introduction

This document assesses the quality and completeness of a project's documentation and website (if present).

This document:

- Measures existing documentation quality against the CNCF’s standards
- Recommends specific and general improvements
- Provides examples of great documentation as reference 
- Identifies key improvements with the largest return on investment

## How this document works

The assessment is divided into three sections: 

- **Project documentation:** for end users of the project; aimed at people who intend to use it
- **Contributor documentation:** for new and existing contributors to the project
- **Website:** branding, website structure, and maintainability

> Each section rates content based on different [criteria](criteria.md).

Author: I think some of the weirdness stems from the fact that [Notary V2 is a spec](https://github.com/notaryproject/notaryproject) right now, with [a prototype cli](https://github.com/notaryproject/nv2), both of which need to be documented (and have been in repo -- but those docs need to be consolidated, merged and brought into a website). In addition, there is a [Notary V1](https://github.com/theupdateframework/notary) out there as well, which also needs to be represented in the website, but which may not be needed for the website MVP (a link to the V1 docs in repo may be enough to start).

 - 3 = (is present, but needs work) 
 - 5 = (is executed extremely well or no improvement required) |

### V1

| Criteria                   | 1   | 2   | 3   | 4   | 5   |
| ---                        | --- | --- | --- | --- | --- |
| Information architecture   |     |     | ✅  |     |     |
| New user content           |     |     | ✅  |     |     |
| Content maintainability    |     | ✅  |     |     |     |
| Content creation processes |     |     | ✅  |     |     |

### V2

| Criteria                   | 1   | 2   | 3   | 4   | 5   |
| ---                        | --- | --- | --- | --- | --- |
| Information architecture   |     |     | ✅  |     |     |
| New user content           | ✅  |     |     |     |     |
| Content maintainability    | ✅  |     |     |     |     |
| Content creation processes | ✅  |     |     |     |     |

### Comments

There are two versions of notary that need to be supported with documentation.
- V1 is housed with [The Update Framework](https://github.com/theupdateframework/notary), and the docs are included as a set of markdown files in the [/docs folder](https://github.com/theupdateframework/notary/tree/master/docs)

> - V2 is housed with The Notary Project (https://github.com/notaryproject/notaryproject) 

Reviewer: Are there two versions of documentation or are there two documentation repositories? 

From what I can see, there are two documentation repositories, and their v2 docs are currently spec/scenario/use case documentation rather than end user documentation.

A recommendation I would make to this project is that they clearly understand that use case documentation is irrelevant to their end user, and that they need to provide documentation for those end users as well.

Author: > Are there two versions of documentation or are there two documentation repositories?

There are 2 versions of Notary, and V2 started new (seemingly) from the ground up. So the documentation versions won't necessarily flow one from the other.
 
> From what I can see, there are two documentation repositories, and their v2 docs are currently spec/scenario/use case documentation rather than end user documentation.

This is correct, V2 docs are primarily specifications. There is a [V2 cli prototype](https://github.com/notaryproject/nv2) with some documentation as well.

**Information Architecture**: 
- As there is no website as yet, and development is still ongoing, V2's Project Documentation is all contained within markdown files in the notaryproject repo.

**New user content**:
- What is there does a good job of explaining the project, but doesn't provide many docs focused on new users.

**Content maintainability**:
- The docs currently aren't versioned or localized, and is depending on GitHub for search.

**Content creation processes**:
- There are links to community discussions, but as there is no website yet there is no space to point to regarding docs creation.

### Recommendations

> V1 Docs will likely need to be reworked once an information architecture is setup for the new site. The new site IA can likely be informed by the existing V1 docs.

Reviewer: In the spirit of "anyone should be able to pick up this work, not just you": spell it out.

"As there is no existing information architecture, we recommend the following, based on the v1 docs: 

- Overview
- Getting Started
- Some lovely concept
   - Some lovely concept or task
   - Another lovely task 
- Another lovely concept  "

.. etc.

> The notaryproject.dev site should primarily focus on the V2 work, but V1 needs to be represented as well.

Reviewer: Should it? The v2 repo is, as discussed above, mostly spec work and internal documentation. I would make the recommendation that before _any_ site work starts, the project team meet with one of CNCF's technical writers/dev advocates to discuss exactly what needs to be documented for end users. I get the sense that they haven't thought about this from a user-centered perspective at all and that there are major gaps in the documentation.

Author: I agree that some more discussion with the maintainers is in order to try to sort out what needs to be documented for the end users -- currently it's hard to see how to start with the V2 project.

I think instead of saying that the notaryproject.dev site should focus on the V2 docs, it may be better to say that _initially_ the site should focus on V2 as a part of an MVP effort. [During my discussion with Steve](https://hackmd.io/2FmXhqxpTFGqM0yEjkmhcw?view), he mentioned that both versions should be represented. So there may be more to discuss regarding what is a priority for building out the site.

Author: I'm trying to keep this assessment and the website separate as they are two projects. But given they are so tightly linked, could the recommendations section include a prioritization of what needs to be done, or should that still only considered in the website project, keeping this assessment purely focused on the documentation tasks that should be done (regardless of when/how)?

> - V1 content seems quite good, but looks like it was developed for a site that no longer uses it.

Reviewer: "quite good?" What specifically does it do well that should be used going forward?

> - For MVP, I'd suggest focusing on V2 content, and linking out to the existing V1 documentation in the location that is currently rather than incorporating it right off the bat.

Reviewer: As a general comment: while I understand that this work is being done in advance of doing a website for the Notary project, your recommendations are so strongly geared _only_ towards what needs to be done that it's _only_ useful for you. 

The purpose of this document is to give _the project_ enough information that they can action on things on their own. Using phrasing like "For MVP", and describing recommendations in vague terms like this doesn't help them achieve that aim.

Author: I agree, the draft you reviewed was written with myself as the audience. Thanks for this feedback, I'm keeping it in mind during my revisions.

> Eventually I'd expect that V1 would be moved to the Notary Project org, but that is a project level consideration. Even if the whole repo doesn't get moved, there may be value in moving the docs out of TUF's notary repo and into the notaryproject/notaryproject.dev repo.

Reviewer: This is a good recommendation, but phrase it as such:

```suggestion
We recommend moving any relevant version 1 content to the notary project org, relieving your users of the need to bounce between two repositories to find information.
```

## Contributor documentation 

**Note**, as my recommendation above is to link to the V1 documentation as is for the website MVP, the remainder of this assessment will only be looking at the existing V2 documentation.

Criteria 
- 1 = (Is not present or requires significant work)
- 3 = (is present, but needs work)
- 5 = (is executed extremely well or no improvement required) |

| Criteria                                  | 1   | 2   | 3   | 4   | 5   |
| ---                                       | --- | --- | --- | --- | --- |
| Communication methods documented          |     |     | ✅  |     |     |
| Beginner friendly issue backlog           |     | ✅  |     |     |     |
| “New contributor” getting started content | ✅  |     |     |     |     |
| Project governance documentation          | ✅  |     |     |     |     |

### Comments

_Provide comments for each rating above, 1-2 sentences max, bullet point list_

> **Communication methods documented**

Reviewer: Once again, the [criteria](https://github.com/cncf/techdocs/blob/main/assessments/criteria.md#communication-methods-documented) are helpful here.

While they do have their communication methods documented, compared to the example provided they could stand to do more :)

- There is a Contributing & Conversations section in the main README (tho, it is mis-linked in the table of contents)

**Beginner friendly issue backlog**
- There are "Documentation," "spec," and "good first issue" issue tags
- Most issues seem to link out to hackmd.io notes, but those note are are well written and informative

**“New contributor” getting started content**
- Same as above, there is a Contributing & Conversations section, but there are not yet any instructions for contributions.

**Project governance documentation**

> - This appears to be actively being worked on, and there is a "Governance" issue tag in use. 

Reviewer: This is being actively worked on but there is no formal governance documentation at this time, which is an issue the project needs to address ASAP.

Author: They've got an open PR on this issue (https://github.com/notaryproject/notaryproject/pull/78/). I will note this in the recommendations.

> ### Recommendations

Reviewer: Given the large gaps in project governance and contributor documentation, one of my major recommendations both in this second *and* for the project as a whole is that they sort that out ASAP. I would mention that a lack of contributor documentation and formal governance will be a huge roadblock to their development as a project and their ability to attract new contributors will be hindered as a result. 

I'm confused as to why this didn't show up as a recommendation for you.

Most of the issues in this section are a function of not having a website yet, and the project still being in a building phase.

> - As we go through the site build project, we should put together a skeleton IA that can be filled in as the project grows.

Reviewer: ```suggestion
- As the website build process starts, put together a skeleton IA that can be filled in as the project grows.
```

> Regarding beginner friendliness, as we move through the project, we should consider processes/documents for:

Reviewer: ```suggestion
Regarding beginner friendliness, consider processes/documents for:
```

In this document, I wouldn't use "we" - please keep in mind that you are there to advise the project and help out with some specific tasks, _not_ to become a permanent member of the project team.

* Triaging docs issues
* Clearly marking a way for new contributors to make code or documentation contributions (i.e. a “good first issue” label), and define what makes a good first issue
* Maintaining issues for staleness
* New contributors/your first contribution
* Letting new users know where to get help

> ## Website

Reviewer: Given that no website exists, I don't think you need to go into too much detail other than saying "No website exists and therefore we will not evaluate this section". And then literally don't fill it in!

- 3 = (is present, but needs work) 
- 5 = (is executed extremely well or no improvement required) |

| Criteria                  | 1   | 2   | 3   | 4   | 5   |
| ---                       | --- | --- | --- | --- | --- |
| Branding and design       |     | ✅  |     |     |     |
| Case studies/social proof | ✅  |     |     |     |     |
| Maintenance planning      | ✅  |     |     |     |     |

### Comments

Currently there is no website, so some of this is not applicable (as we're starting the project of building one)

**Branding and design**:
- Artwork exists https://github.com/cncf/artwork/tree/master/projects/notary
 
**Case studies/social proof**:
- I'm not sure this is applicable for V2 yet. V1 may have some of this information.

**Maintenance planning**: 
- N/A

### Recommendations

There may not be much now, but as we collect information for this site, we can ask the following questions:

**Regarding Case studies/social proof**:
* Are there case studies available for the project?
* Are there user testimonials available?
* Can we start a blog? 
* Are there community talks for the project 
* Is there a logo wall of users/participating organizations?

> ## Final Recommendations

Reviewer: For this section, please see [what I did for project Contour](https://github.com/cncf/techdocs/blob/main/assessments/0001-contour.md#final-recommendations).

Please also refer to the [instructions in the template](https://github.com/cncf/techdocs/blob/main/assessments/template.md#recommendations):

> From the recommendations above, lis the top 1-3 concerns for this particular project and expand on them in enough detail that you could either: - Pass the work off to a contractor or other member of the CNCF techdocs team - Write a GitHub issue for the project team and place it in the backlog and someone not involved in the docs assessment process could execute on it

_Each issue in this section_ should be, for lack of a better word, a fully formed GitHub issue or even an umbrella issue. 

I'm not going to review what you've written here in detail, because I think this section needs to be rewritten.

## How this document works

This assessment is divided into two sections: 

- **Project documentation:** for end users of the project; aimed at people who intend to use it
- **Contributor documentation:** for new and existing contributors to the project

Each section rates content based on different [criteria](criteria.md).

Normally, a **website** section would also be a part of a document assessment, but as there isn't a website for the Notary Project yet, the website assessment section will be skipped, but there are some website recommendations in the [Final Recommendations](#final-recommendations) section. 

## Project documentation 

| Criteria                   | 1   | 2   | 3   | 4   | 5   |
| ---                        | --- | --- | --- | --- | --- |
| Information architecture   |     | ✅  |     |     |     |
| New user content           | ✅  |     |     |     |     |
| Content maintainability    | ✅  |     |     |     |     |
| Content creation processes | ✅  |     |     |     |     |

 Criteria: 
 - 1 = (Is not present or requires significant work) 
 - 3 = (is present, but needs work) 

>  - 5 = (is executed extremely well or no improvement required) |

Author: Will do :)

   - [Notary V2 prototype cli](https://github.com/notaryproject/nv2)
- The [V1 Notary Overview](https://github.com/theupdateframework/notary#overview) is reasonable and explanatory.
- The [Notary V2 Overview](https://github.com/notaryproject/notaryproject#notary-v2-overview) provides a good explanation of the V2 project, but which is aimed at the specification writers/developers.

- Is the documentation feature complete? (i.e., each product feature is documented)
   - As the project is in ongoing development, not every feature is completed and so there are some holes in the documentation.
   - The https://github.com/notaryproject/nv2 repository contains the nv2 prototype cli as well as the documentation about how to use it. Again, since the project is in development it seems like there are some holes. 

- Are there step-by-step instructions (tasks, tutorials) documented for features?
   - Since V2's documentation is so conceptual at this point (consisting mostly of specification), it may be best to leave it as is -- linking to it directly during the development process, and planning documentation along side the development of the system as a whole.

> - Is the “happy path”/most common use case documented.

Author: We may want to check the criteria file for this as well - I think i may have grabbed it verbatim from there.

   - The V1 getting started guide is very good, will V2 be able to draw from it?

> - If the documentation does not suffice, is there a clear escalation path for users needing more help? (FAQ, Troubleshooting)

Author: (I think this is from the critera file as well)

   - This is an area for improvement on both V1 and V2 docs. Potentially consider reorganizing the whole docset with users in mind: dividing the content in a concepts, tasks, and tutorials model. 

#### New user content

- Getting Started Guide is clearly labelled: https://github.com/theupdateframework/notary#getting-started-with-the-notary-cli though notably only in v1 of their docs, not v2. How does one get started with v2?

> - The v2 overview has getting started/CLI steps for an entirely different set of functions: https://github.com/notaryproject/notaryproject#notary-v2-overview. Are these in addition to or in replacement of the v1 getting started steps?

Reviewer 2: Live links here might be helpful.

- Where do users go after the getting started steps? Is there anything else about the tool they need to understand?
- Sample code looks great!

#### Content maintainability

> - Regarding searchability, as the documentation is all in GitHub currently, it is searchable; however, as a part of setting up a website, we recommend using a tool like Google, Algolia, or Lunr to provide search functionality. 

Reviewer: ```suggestion
- As the documentation in GitHub it is searchable. When setting up a website, we recommend using Google, Algolia, or Lunr to provide search functionality. 
```

When possible, use less words.

Author: Good advice -- I do tend to verbosity.

  
- Is the Notary Project planning on localization/internationalization? If so, we recommend starting the site with a directory structure which supports localization (Hugo supports this very well out of the box).

> - Regarding versioning, does V1 lead directly to V2?

Reviewer: Be careful of your use of the word "regarding". It's a "crutch" word for you.

Author: Good to know, thanks for the heads up!

  - If so, import v1 into the site, and "version" by copying the content into a new directory for v2 and updating only what needs to change or be added.

>   - If not, upload v1 and v2 into the site as separate document sets. 

Reviewer 2: When referring to versions, a consistent way of writing "version", "v#", or "V#" helps reduce cognitive load, teaching the pattern with the first instance. (see line 82 vs lines 83/84)

> - Regarding versioning, does V1 lead directly to V2?
>    - If not, then V1 and V2 should be brought in as separate document sets, and V2 should be set up with a versioning process (allowing V1 to exist as a sort of archive).
>    - If so, then V1 can act as a base for the site and no separation of versions need be applied.

Reviewer: ```suggestion
- Is any of the content from version 1 relevant to version 2? 
  - If so, import v1 into the site, and "version" by copying the content into a new directory for v2 and updating only what needs to change or be added.
  - If not, upload v1 and v2 into the site as separate document sets. 
```

I think this is what you're trying to say?

Author: Yes, I think this is clearer.

#### Content creation processes

<!-- borrowed from @celestehorgan's excellent work in 0001-contour.md -->
- The following are ideas to consider going forward with a docs repo. As a content creator:
   - If I write documentation, who verifies that it's technically accurate? 
   - If I'm not a native english speaker, who can I ask for help with grammar and language review? 
   - If I'm a trained technical writer and want to rearrange, create new, or split existing topics, who do I tag in issues/on Slack for comment?
   - As a developer user, how can I ensure that content is accurate and up to date? Are alpha and beta features flagged? What happens when content (+ functionality) gets deprecated or changed in a major way? 
   - Who reviews documentation PRs? 

### Recommendations

#### Information Architecture

The assumption is that V2 is the primary project moving forward, and that a Minimum Viable Product (MVP) for a website/documentation will focus on Notary V2 users.

> - Bring the nv2 cli prototype documentation into the main documentation when conosolidating the content on the website.

Reviewer: ```suggestion
- Bring the Notary v2 CLI prototype documentation into the main documentation when consolidating the content on the website.
```

1. Do not introduce abbreviations and other shorthand
2. Capitalize CLI and other acronyms 
3. Define "main documentation" for the reader (I can't do this for you, as I'm honestly not sure what you're referring to here)

> - V1 Docs will need to be reworked once an information architecture (IA) is setup for the new site. The new site IA can likely be informed by the existing V1 docs.

Reviewer: ```suggestion
- Rework the v1 Docs  once an information architecture (IA) is set up for the new site. Set up the new site IA based on the structure of the v1 docs unless an alternative is proposed.
```

> The [Service Mesh Interface website](https://smi-spec.io/) handles its specification tab very well -- linking directly to the relevant markdown folder in GitHub, allowing website users to find the specification work where it is happening, and giving new users an opportunity to participate.

Reviewer: This isn't a recommendation on its own.

Try:

```suggestion
- Separate specifications from user-centered documentation. For example, how [Service Mesh Interface website](https://smi-spec.io/) handles its specification.
```

## Contributor documentation 

| Criteria                                  | 1   | 2   | 3   | 4   | 5   |
| ---                                       | --- | --- | --- | --- | --- |
| Communication methods documented          |     |     | ✅  |     |     |
| Beginner friendly issue backlog           |     | ✅  |     |     |     |
| “New contributor” getting started content | ✅  |     |     |     |     |
| Project governance documentation          | ✅  |     |     |     |     |

Criteria 
- 1 = (Is not present or requires significant work)
- 3 = (is present, but needs work)
- 5 = (is executed extremely well or no improvement required) |

> ### Comments & recommendations

Reviewer: Avoid changing a templated document unless you have good reason to. In this case I'll let it slide.

Author: Now I'm confused. I don't think I changed the content here, just the formatting, same as above: https://github.com/cncf/techdocs/pull/27/files#r682954996

Author: I think this improves the readability enough to be a good reason to make the change.

> - Most of the issues in this section are a function of not having a website yet, and the project still being in a building phase.

Reviewer: ```suggestion
- Most of the issues in this section are a function of not having a website yet.
```

Most projects are still in a building phase, we can leave it here.

  - As the website build process starts, put together a skeleton IA that can be filled in as the project grows.

> - There are documented community communication methods, but it could be collected and expanded during the site development process. Prometheus has a good example of a community page: https://prometheus.io/community/. 

Reviewer: ```suggestion
- Communication methods are documented but could be further centralized and expanded during the site development process, with an end result comparable to https://prometheus.io/community/. 
```

In this sentence:
1. "it could be collected" -> the "it" pronoun is incorrect, as "community communication methods" is plural.
2. "there are documented community communication methods" is a bit awkwardly phrased. 

Try the suggestion above.

> - Regarding beginner friendliness, consider processes/documents for:

Reviewer: Once again with the "Regarding"

Reviewer: ```suggestion
- There is no documentation for:
```

## Final Recommendations

Is it possible plan documentation along side the V2 development? Will the documentation be able to get its structure from the specifications being developed?

### Website

- [x] Create separate documentation repository: https://github.com/notaryproject/notaryproject.dev
- [ ] Setup a [Hugo website](https://gohugo.io) with a [Docsy theme](https://www.docsy.dev/)
- [ ] Setup Netlify for hosting/CI
- [ ] Develop a [maintainers list](https://github.com/notaryproject/notaryproject.dev/blob/main/.github/settings.yml) for the documentation and its repository (this has been started https://github.com/notaryproject/notaryproject/issues/77)
- [ ] Develop a maintenance plan for the documentation and its repository 
- [ ] Follow the [CNCF website guidelines checklist](https://github.com/cncf/techdocs/blob/main/howto/website-guidelines-checklist.md) for other required elements

_Note_: [Notary Project branding artwork](https://github.com/cncf/artwork/tree/master/projects/notary) exists. Other branding elements, like colour schemes and whatnot, will need to be developed.

### Information Architecture

> The content to be written could have the following structure (as a start - modifications will need to be made as development continues):

Reviewer: ```suggestion
We recommend reorganizing your content in the following way: 
```

As a general note: your writing is very tentative and shy. Speak with more authority, in active voice and present tense. 

Common phrases for you are:
- "The ___ to be ___..." (future tense, we prefer present tense in technical writing) 
- "..could have.... as a start" (You are making a recommendation as an expert. There is no "could" or "as a start". Things may change as you go through the project, but when giving people your expertise speak with confidence and finality.
- ("..will need") -> passive voice, we prefer active voice in technical writing. An easy way to test for passive voice is to add "... by dogs" at the end of the sentence. If it still makes sense, you're speaking in passive voice. For example:

"modifications will need to be made... by dogs" -> Bingo, passive voice!
versus
"As development continues this plan might change... by dogs" -> grammatically incorrect, we're speaking in active voice!

There's also no need to caveat that things might change as you go through development. This is an initial recommendation/plan - some modification as you progress is expected.

**Site structure**

- Landing page
- Community
   - Governance
   - Community (details)
   - Contributing
      - Where to start
      - How we work
      - Filing and working on issues for the project
      - Where the community is (contacts, meetings)
      - Contributing to docs
      - Where to ask for more help contributing
      - Next steps
- About
- Blog (Get the word out about the project and how it's progressing)
- Docs
   - This section to be planned with the Notary Maintainers team

>    - Getting Started

Author: I had considered that -- but I was thinking ahead to any potential versioning. I think a Getting Started guide should be organized within the versioned portion of a site as it could change with project updates. A Getting Started guide within the docs folder can (and should) still be referenced on the home page, and is in my **Top Nav** suggestion below.

>    - Getting Started

Reviewer: For now let's leave it and see how it plays out in the final website.

      - This section to be planned with the Notary Maintainers team

> **Topnav**

Reviewer: ```suggestion
**Top nav**
```

- Docs/Specification (Specification initially, then Docs as development progresses)
- Version 1 Docs (https://github.com/theupdateframework/notary/tree/master/docs initially, then /docs/v1/(latest?))
- Community
- Getting started
- Specification (https://github.com/notaryproject/notaryproject initially)

### Moving and consolidating existing documentation

> - We recommend that the V1 Notary Project that is currently housed in the [The Update Framework](https://github.com/theupdateframework/notary/) GitHub organization be moved into the [Notary Project](https://github.com/notaryproject) GitHub organization.

Reviewer: ```suggestion
- We recommend moving the v1 Notary Project project from the [The Update Framework](https://github.com/theupdateframework/notary/) GitHub organization to the [Notary Project](https://github.com/notaryproject) GitHub organization.
```

A  good way to proofread your own work is to look through it and see if there are words you can remove or combine into one wor.d For example, 

"that is currently housed" -> "that is in", "in" both work. 

In this case, rearranging the sentence to use active voice (moved -> moving) removes lots of the extra words.

> - The V1 documentation makes reference to Docker (which was the original home of the documentation). As the V1 docs are moved from their original location, references to things like "the Docker Notary documenation" should be removed.

Reviewer: ```suggestion
- Remove any references to Docker (the original home of  the documentation), for example "the Docker Notary documentation".
```
This is once again an active voice issue:

"references to things.... should be removed (by dogs)" -> passive
"Remove any references to.... for example.... (by dogs)" -> active

- Even if the V1 Notary repo isn't moved, the website MVP should link to V1 notary where it is now, with no changes.
- Docs+Code combined repositories are a long-term risk. We strongly recommend decoupling these into their own repositories

>    - As the website is being built, we recommend the existing V2 documentation be moved out of the https://github.com/notaryproject/notaryproject repo, and into the new documentation repository: https://github.com/notaryproject/notaryproject.dev

Author: I agree with the suggestion to use active voice, in this case (and throughout the doc) I was trying to avoid "we" except in reference to CNCF recommendations (and even then I'm not super keen on "we"). I'll see if I can rewrite this line to both use active voice, and avoid "we."

This assessment is divided into two sections: 

- **Project documentation:** for end users of the project; aimed at people who intend to use it
- **Contributor documentation:** for new and existing contributors to the project

Each section rates content based on different [criteria](criteria.md).

Normally, a **website** section would also be a part of a document assessment, but as there isn't a website for the Notary Project yet, the website assessment section will be skipped, but there are some website recommendations in the [Final Recommendations](#final-recommendations) section. 

## Project documentation 

| Criteria                   | 1   | 2   | 3   | 4   | 5   |
| ---                        | --- | --- | --- | --- | --- |
| Information architecture   |     | ✅  |     |     |     |
| New user content           | ✅  |     |     |     |     |
| Content maintainability    | ✅  |     |     |     |     |
| Content creation processes | ✅  |     |     |     |     |

 Criteria: 
 - 1 = (Is not present or requires significant work) 
 - 3 = (is present, but needs work) 
 - 5 = (is executed extremely well or no improvement required) |

### Comments

#### Information Architecture

> 

Reviewer: This section looks much better and more thorough! 💖

#### Content creation processes

<!-- borrowed from @celestehorgan's excellent work in 0001-contour.md -->
- The following are ideas to consider going forward with a docs repo. As a content creator:
   - If I write documentation, who verifies that it's technically accurate? 
   - If I'm not a native english speaker, who can I ask for help with grammar and language review? 
   - If I'm a trained technical writer and want to rearrange, create new, or split existing topics, who do I tag in issues/on Slack for comment?
   - As a developer user, how can I ensure that content is accurate and up to date? Are alpha and beta features flagged? What happens when content (+ functionality) gets deprecated or changed in a major way? 
   - Who reviews documentation PRs? 

### Recommendations

#### Information Architecture

The assumption is that V2 is the primary project moving forward, and that a Minimum Viable Product (MVP) for a website/documentation will focus on Notary V2 users.
- Bring the nv2 cli prototype documentation into the main documentation when conosolidating the content on the website.

> - V1 Docs will need to be reworked once an information architecture is setup for the new site. The new site IA can likely be informed by the existing V1 docs.

Reviewer 2: Asking genuinely – what does IA mean?

Reviewer: "Information Architecture" or, basically: "what order is the content in, where are the topics/subtopics, and is that ordering/grouping good?"

Reviewer 2: Thanks, Celeste! @nate-double-u what do you think of specifying what acronyms stand for the first time we use them? (I frequently forget this myself.)

Author: > Thanks, Celeste! @nate-double-u what do you think of specifying what acronyms stand for the first time we use them? (I frequently forget this myself.)

Good suggestion, I'll do that

> The [Service Mesh Interface website](https://smi-spec.io/) handles it's specification tab very well -- linking directly to the relevant markdown folder in GitHub, allowing website users to find the specification work where it is happening, and giving new users an opportunity to participate.

Reviewer: ```suggestion
The [Service Mesh Interface website](https://smi-spec.io/) handles its specification tab very well -- linking directly to the relevant markdown folder in GitHub, allowing website users to find the specification work where it is happening, and giving new users an opportunity to participate.
```

its = possessive (the dog and its house)
it's = a contraction of "it is" (it's a house)

Author: Good catch, thx!
(typo 😅)

  - As the website build process starts, put together a skeleton IA that can be filled in as the project grows.

- There are documented community communication methods, but it could be collected and expanded during the site development process. Prometheus has a good example of a community page: https://prometheus.io/community/. 

- Regarding beginner friendliness, consider processes/documents for:
   - Triaging docs issues
   - Clearly marking a way for new contributors to make code or documentation contributions (i.e. a “good first issue” label), and define what makes a good first issue
   - Set up a process for maintaining/pruning stale issues
   - New contributors/your first contribution
   - Letting new users know where to get help

- Complete the governance work started here: https://github.com/notaryproject/notaryproject/issues/55 https://github.com/notaryproject/notaryproject/pull/78, and add it to a governance page (or section) on the website.

## Final Recommendations

Is it possible plan documentation along side the V2 development? Will the documentation be able to get its structure from the specifications being developed?

> ### Website

Reviewer 2: I love the checklist format of this section. It's concise and actionable.

Reviewer: +1! This is an awesome way to do this part. I'm glad you did this, @nate-double-u - it's a good model of what we can do with these going forward.

Author: Thanks -- should I refactor and use this format in other areas of this document?

_Note_: [Notary Project branding artwork](https://github.com/cncf/artwork/tree/master/projects/notary) exists. Other branding elements, like colour schemes and whatnot, will need to be developed.

### Information Architecture

The content to be written could have the following structure (as a start - modifications will need to be made as development continues):

**Site structure**

- Landing page
- Community
   - Governance
   - Community (details)
   - Contributing
      - Where to start
      - How we work
      - Filing and working on issues for the project
      - Where the community is (contacts, meetings)
      - Contributing to docs
      - Where to ask for more help contributing
      - Next steps
- About
- Blog (Get the word out about the project and how it's progressing)
- Docs
   - This section to be planned with the Notary Maintainers team
   - Getting Started
      - This section to be planned with the Notary Maintainers team

**Topnav**

- Docs/Specification (Specification initially, then Docs as development progresses)
- Version 1 Docs (https://github.com/theupdateframework/notary/tree/master/docs initially, then /docs/v1/(latest?))
- Community
- Getting started

> - Specification (https://github.com/notaryproject/notaryproject initially)

Reviewer: 👍 👍 👍  Great work

Author: Thanks!
